pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -63,6 +64,24 @@ func (src *EnvironmentResource) ConvertTo() (v1.DataModelInterface, error) {
 	}
 	converted.Properties.Compute = *envCompute
 
+	recipeConfig := src.Properties.RecipeConfig
+	if recipeConfig != nil {
+		if recipeConfig.Terraform != nil {
```

Comment:
I agree with @sk593. We can have a function that takes care of the recipeConfig conversion and that function can make a call to another function which takes care of terraform related stuff conversion. With this approach we could keep things simple. It would be more readable, easier to debug and write unit tests.

Reply:
done